src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -617,14 +624,28 @@ internal static class Shards
             }
         }
 
-        public static void PassivateCompleted<TShard>(this TShard shard, Shard.EntityStopped evt) where TShard: IShard
+        private static void PassivateIdleEntities<TShard>(this TShard shard) where TShard : IShard
```

Comment:
My first implementation was easier to the eye and more in line with the JVM, but probably less _performant_:

```csharp
var deadline = DateTime.Now.Ticks - shard.Settings.PassivateIdleEntityAfter.Ticks;
var refsToPassivate = shard.LastMessageTimestamp.Where(s => s.Value < deadline).Select(s => shard.RefById[s.Key]).ToArray();
if (refsToPassivate.Any())
{
    shard.Log.Debug($"Passivating [{refsToPassivate.Length}] idle entities");
    refsToPassivate.ForEach(entity => Passivate(shard, entity, shard.HandOffStopMessage));
}
```

Reply:
👍